qiskit/ignis/experiments/calibration/instruction_data/utils.py

```diff
@@ -12,211 +12,90 @@
 
 """Utilities for database."""
 
-from copy import deepcopy
+import inspect
+import re
 from enum import Enum
-from typing import Optional, Tuple, Dict
+from typing import Dict, List
 
-import numpy as np
-from qiskit.qobj.converters.pulse_instruction import ParametricPulseShapes
+from qiskit import pulse
 
-from qiskit import pulse, circuit
-from qiskit.ignis.experiments.calibration.exceptions import CalExpError
-from qiskit.ignis.experiments.calibration.instruction_data.database import (PulseTable,
-                                                                            ScheduleTemplate)
 
+class ParametricPulseShapes(Enum):
+    """Map the pulse shape name to the pulse module waveforms.
+
+    The enum name is the DSL name for pulse shapes, the
+    value is its mapping to the OpenPulse Command in Qiskit.
+    """
+    gaus = pulse.Gaussian
+    gaus_sq = pulse.GaussianSquare
+    drag = pulse.Drag
+    constant = pulse.Constant
```

Comment:
Say I, as a user, have created my own `ParametricPulse` e.g. `class MyPulse(ParametricPulse)`, how do I register it into this enum?

Reply:
You can input the Enum class to inst def now.
https://github.com/nkanazawa1989/qiskit-ignis/blob/2aac632193c25d32696c7dfd97d22d9c787c8528/qiskit/ignis/experiments/calibration/instruction_data/gate_and_schedules_definition.py#L45-L46

Reply:
Right, so a user can define his own Enum class? In which case that should be fine.